src/lints/struct_field_marked_deprecated.ron

```diff
@@ -0,0 +1,66 @@
+SemverQuery(
+    id: "struct_field_marked_deprecated",
+    human_readable_name: "pub struct field is now #[deprecated]",
+    description: "A pub struct field is now marked #[deprecated] and will be removed in a future version.",
+    required_update: Minor,
+    lint_level: Deny,
+    reference_link: Some("https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute"),
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Struct {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+
+                        importable_path {
+                            path @output @tag
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        field {
+                            field_name: name @output @tag
+                            public_api_eligible @filter(op: "=", value: ["$true"])
+                            deprecated @filter(op: "!=", value: ["$true"])
+                        }
+                    }
+                }
+            }
+            current {
+                item {
+                    ... on Struct {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        struct_name: name @output
+
+                        # It's possible that both the struct and its field became deprecated.
+                        # This is a rare case where we don't consider the lint on the field
+                        # to be a duplicate lint of the one on the struct.
+                        #
+                        # Deprecation on the struct means we shouldn't use it in new code,
+                        # but existing code can still access its fields until they are removed.
```

Comment:
Consider adding some test cases that involve `#[doc(hidden)]` on both the struct and its fields, to pin down the exact behavior of this lint.

This is especially important when leaving a comment describing some non-obvious decision made in the lint, like this one here.